File: leetcode/0001. Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        """
        Using hashmap to log missing parts (target - nums[index]).
        If we already find a number that satisfies this question, 
        we can return it.
        """
        hash_map = dict()
        for i, e in enumerate(nums):
            if e in hash_map.keys(): return [i, hash_map[e]]
            else: hash_map[target - e] = i
        return []

File: leetcode/0001. Two Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        /*
         * Using hashmap to log missing parts (target - nums[index]).
         * If we already find a number that satisfies this question,
         * we can return it.
         */
        HashMap<Integer, Integer> map = new HashMap();

        for (int i = 0; i < nums.length; i++) {
            if (map.containsKey(nums[i]))
                return new int[] { i, map.get(nums[i]) };
            else
                map.put(target - nums[i], i);
        }

        return new int[] { 0, 0 };
    }
}